ansible/roles/setup_mysql/tasks/main.yml

```diff
@@ -3,14 +3,29 @@
 
 - name: Install mysql-server
   apt: name=mysql-server update_cache=yes cache_valid_time=36000 state=present
-- name: Set password for root
-  command: mysqladmin -u root password {{mysql_root_password}}
-  ignore_errors: yes
+- name: Update mysql root password for all root accounts
+  mysql_user: name=root host=localhost password={{mysql_root_password}}
+- name: Copy .my.cnf file with root password credentials
+  template: src=templates/root/.my.cnf dest=/root/.my.cnf owner=root mode=0600
+- name: Update mysql root password for all root accounts
+  mysql_user: name=root host={{ item }} password={{mysql_root_password}}
+  with_items:
+    - "{{ansible_hostname}}"
+    - 127.0.0.1
+    - ::1
```

Comment:
This properly sets up the root user and is the recommended way in the ansible docs (requires the .my.cnf template)